persistent-postgresql/Database/Persist/Postgresql/Internal/Migration.hs

```diff
@@ -1097,80 +1097,87 @@ findAlters
+                                    Just s -> (:) (UpdateNullToValue newCol s)
                              in
-                                up [NotNull col]
+                                up [NotNull newCol]
                         _ -> []
                     modType
-                        | sqlTypeEq sqltype sqltype' = []
+                        | sqlTypeEq (cSqlType newCol) (cSqlType oldCol) = []
                         -- When converting from Persistent pre-2.0 databases, we
                         -- need to make sure that TIMESTAMP WITHOUT TIME ZONE is
                         -- treated as UTC.
-                        | sqltype == SqlDayTime && sqltype' == SqlOther "timestamp" =
-                            [ ChangeType col sqltype $
+                        | cSqlType newCol == SqlDayTime && cSqlType oldCol == SqlOther "timestamp" =
+                            [ ChangeType newCol (cSqlType newCol) $
                                 T.concat
                                     [ " USING "
-                                    , escapeF name
+                                    , escapeF (cName newCol)
                                     , " AT TIME ZONE 'UTC'"
                                     ]
                             ]
-                        | otherwise = [ChangeType col sqltype ""]
+                        | otherwise = [ChangeType newCol (cSqlType newCol) ""]
                     modDef =
-                        if def == def'
-                            || isJust (T.stripPrefix "nextval" =<< def')
+                        if cDefault newCol == cDefault oldCol
+                            || isJust (T.stripPrefix "nextval" =<< cDefault oldCol)
                             then []
-                            else case def of
-                                Nothing -> [NoDefault col]
-                                Just s -> [Default col s]
+                            else case cDefault newCol of
+                                Nothing -> [NoDefault newCol]
+                                Just s -> [Default newCol s]
                     dropSafe =
-                        if safeToRemove edef name
-                            then error "wtf" [Drop col (SafeToRemove True)]
+                        if safeToRemove edef (cName newCol)
+                            then error "wtf" [Drop newCol (SafeToRemove True)]
                             else []
                  in
                     ( modRef ++ modDef ++ modNull ++ modType ++ dropSafe
-                    , filter (\c -> cName c /= name) cols
+                    , filter (\c -> cName c /= cName newCol) oldCols
                     )
+  where
+    refAdd Nothing = []
+    -- This check works around a bug where persistent will sometimes
+    -- generate an erroneous ForeignRef for ID fields.
+    -- See: https://github.com/yesodweb/persistent/issues/1615
+    refAdd _ | fmap fieldDB (getEntityIdField edef) == Just (cName newCol) = []
```

Comment:
This guard is basically the same as it was before - before, we were checking against the old column's name, and now we're checking against the new column's name, because there won't always be an old column name to check against (eg in the event that the column doesn't yet exist). There's no meaningful difference - previously, we would only ever run this check after verifying that the old column name matched the new column name.